app/src/main/java/com/duckduckgo/app/browser/omnibar/experiments/FadeOmnibarLayout.kt

```diff
@@ -96,78 +91,103 @@ class FadeOmnibarLayout @JvmOverloads constructor(
         } else {
-            val isScrollingDown = scrollY > previousScrollY
-            previousScrollY = scrollY
-
-            if (isScrollingDown) {
-                targetToolbarAlpha = 0f
-                targetTextAlpha = 1f
-                targetHeight = minibar.height
-            } else {
-                targetToolbarAlpha = 1f
-                targetTextAlpha = 0f
-                targetHeight = toolbar.height
-            }
-
-            // top of the page condition
-            if (scrollY == 0) {
-                revealToolbar()
-            }
-
-            // smooth alpha transition
-            val toolbarAlphaDifference = (targetToolbarAlpha - currentToolbarAlpha)
-            val toolbarAlphaChange = (toolbarAlphaDifference * 0.1f)
-            currentToolbarAlpha += toolbarAlphaChange
-            fadeToolbar(currentToolbarAlpha)
-
-            val textAlphaDifference = (targetTextAlpha - currentTextAlpha)
-            val textAlphaChange = (textAlphaDifference * 0.1f)
-            currentTextAlpha += textAlphaChange
-            fadeMinibar(currentTextAlpha)
-
-            // Calculate the new height for the AppBarLayout
-            val layoutParams = layoutParams
-
-            if (currentHeight == 0) {
-                currentHeight = layoutParams.height
-            }
-
-            if (targetHeight != currentHeight) {
-                updateLayoutHeight()
-            }
+            targetToolbarAlpha = 1f
+            targetTextAlpha = 0f
+        }
+
+        val toolbarAlphaDifference = (targetToolbarAlpha - currentToolbarAlpha)
+        val toolbarAlphaChange = (toolbarAlphaDifference * transitionStepRatio)
+        currentToolbarAlpha += toolbarAlphaChange
+        currentToolbarAlpha = currentToolbarAlpha.coerceIn(0f, 1f)
+        fadeToolbar(currentToolbarAlpha)
+
+        val textAlphaDifference = (targetTextAlpha - currentTextAlpha)
+        val textAlphaChange = (textAlphaDifference * transitionStepRatio)
+        currentTextAlpha += textAlphaChange
+        currentTextAlpha = currentTextAlpha.coerceIn(0f, 1f)
+        fadeMinibar(currentTextAlpha)
+
+        // update layout height
+        targetHeight = if (isScrollingDown) {
+            minibar.height
+        } else {
+            toolbar.height
+        }
+        if (targetHeight != currentHeight) {
+            updateLayoutHeight(transitionStepRatio)
         }
     }
 
     private fun revealToolbar() {
+        // update layout alpha
         targetToolbarAlpha = 1f
         targetTextAlpha = 0f
         currentToolbarAlpha = 1f
         currentTextAlpha = 0f
-        previousScrollY = Int.MAX_VALUE
+        fadeToolbar(targetToolbarAlpha)
+        fadeMinibar(targetTextAlpha)
+
+        // update layout height
+        targetHeight = toolbar.height
+        currentHeight = toolbar.height
+        layoutParams.height = targetHeight
+        this.layoutParams = layoutParams
+    }
+
+    private fun hideToolbar() {
```

Comment:
nit: rename to revealMinibar

Reply:
Addressed in https://github.com/duckduckgo/Android/pull/5782.